File: ssr_pkg/scripts/chao_node.py
#!/usr/bin/env python3
#coding=utf-8
import rospy
from std_msgs.msg import String

if __name__== "__main__":
  rospy.init_node('chao_node')
  rospy.logwarn("This is a warning message")

  pub = rospy.Publisher('chao_node', String, queue_size=10)

  rate = rospy.Rate(10) # 10hz

  while not rospy.is_shutdown():
    rospy.loginfo("This is an info message")
    msg = String()
    msg.data = "Hello, world!"
    pub.publish(msg)
    rate.sleep()

File: ssr_pkg/src/chao_node.cpp
#include<ros/ros.h>
#include<std_msgs/String.h>
#include<qq_msgs/Carry.h>//自定义消息

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "chao_node");
    printf("你已经成功编译了一个ROS节点\n");

    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<qq_msgs::Carry>("chao_node", 10);//发布话题
    ros::Rate loop_rate(10);
    while(ros::ok())
    {
       printf("你已经成功编译了一个ROS节点！\n");
       qq_msgs::Carry msg;
       msg.grade = "king";//自定义消息
       msg.star =50;//自定义消息
       msg.data = "Hello, ROS!";
       pub.publish(msg);
       loop_rate.sleep();
    }
    return 0;
}